faux-mgs/src/main.rs

```diff
@@ -1457,6 +1502,79 @@ async fn monorail_unlock(
         UnlockChallenge::Trivial { timestamp } => {
             UnlockResponse::Trivial { timestamp }
         }
+        UnlockChallenge::EcdsaSha2Nistp256(data) => {
+            let keys = ssh_list_keys()?;
+            let pub_key = if keys.len() == 1 {
+                keys[0].clone()
```

Comment:
Kinda nitpicky, but if `pub_key` is `Some(_)` should we check that it matches the one key in this case?

Reply:
Good catch, changed to `if keys.len() == 1 && pub_key.is_none()`